channel/consensus_channel/consensus_channel.go

```diff
@@ -204,22 +192,42 @@ func (a Add) RightDeposit() big.Int {
 
 var ErrIncorrectTurnNum = fmt.Errorf("incorrect turn number")
 
-// Add updates Vars by including a guarantee, updating balances accordingly
-func (vars Vars) Add(p Add) (Vars, error) {
+// Add mutates Vars by
+// - increasing the turn number by 1
+// - including the guarantee
+// - adjusting balances accordingly
+func (vars *Vars) Add(p Add) error {
 	if p.turnNum != vars.TurnNum+1 {
-		return Vars{}, ErrIncorrectTurnNum
+		return ErrIncorrectTurnNum
 	}
 
+	// Increase the turn number
 	vars.TurnNum += 1
 
-	o, err := vars.Outcome.DivertToGuarantee(p)
+	o := vars.Outcome
+	g := p.Guarantee
```

Comment:
This doesn't seem necessary. Fields of the embedded struct can be accessed directly on `p` like `p.target`.

Reply:
a44421f4f76e551b0192e4d82d6e806256c23347